### PR TITLE
ADR on email sending and delivery responsibilities

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/Xd47oJKZ/535-remove-the-deliveryattempt-model

This reflects a number of recent changes made to the system with how it
communicates with Notify.

👉 [Rendered](https://github.com/alphagov/email-alert-api/blob/email-sent-adr/docs/adr/adr-005-email-delivery-responsibilities.md)

### CODE REVIEW

File: docs/adr/adr-005-email-sending-and-delivery-responsibilities.md
Comment (benthorner) [inline docs/adr/adr-005-email-sending-and-delivery-responsibilities.md:48]:
```suggestion
where an email is in a transitory `pending` status that could mean anything from not sent, to delivered but the callback got lost. We will consider it the
```

File: docs/adr/adr-005-email-sending-and-delivery-responsibilities.md
Comment (benthorner) [inline docs/adr/adr-005-email-sending-and-delivery-responsibilities.md:58]:
```suggestion
Email Alert API would consider a sent email to be in a `pending` status until
```

This isn't quite true, if we couldn't send it to Notify, or the delivery failed (depending on what period in time we're thinking about).

**Review by benthorner — COMMENTED** [review]
Not a grammatical error in sight 💯. It's also a helpful RFC, and thanks for putting effort into the extensive discussion about _sending_ vs. _delivery_, as that's quite a subtle distinction that is likely the origin of the mess we were in!

I've got a few comments, but they're mostly just suggestions on two themes: 

- There are two tenses in use, which was confusing when I read this from top-to-bottom. I think it's most useful to write in the past tense, as then it makes sense to see some links to PRs and commits, which will help future readers follow along.

- The word "send" is used in a few contexts, and sometimes I was confused about the scope of what it's referring to. We could try and define the scope for "send", or maybe be consistently specific e.g. "send to Notify" vs. "send all the way to end user".

File: docs/adr/adr-005-email-sending-and-delivery-responsibilities.md
Comment (kevindew) [reply docs/adr/adr-005-email-sending-and-delivery-responsibilities.md]: Thanks - I've flipped that a bit. That's me struggling with the rather unusual ADR format we have here (that I'm pretty sure I'm to blame for 🤦 )

File: docs/adr/adr-005-email-sending-and-delivery-responsibilities.md
Comment (kevindew) [reply docs/adr/adr-005-email-sending-and-delivery-responsibilities.md]: This seemed an unnecessary detail - particularly as we think archiving isn't used often.

**Comment by kevindew** [comment]
Thanks @benthorner - I've addressed your comments. I put the changes in a separate commit for easier reviewing but will squash before merge.

**Review by benthorner — COMMENTED** [review]
Thanks @kevindew! I've had another read through, and it was easier this time so thanks for making those changes. There's still one more major comments about the archiving, which I didn't connect with this work initially. I also struggled a bit with the structure still, and think there are few simple changes that would make it clearer and reduce some repetition:

- With only a few movements, we could re-partition the paragraphs into "Context", "Decision" and "Consequences" sections, which I've found more helpful in [the ADRs for Content Publisher](https://github.com/alphagov/content-publisher/blob/master/docs/adr/0001-record-architecture-decisions.md). Let's depart from the old layout if it leads to confusing results! We could make ADR 5 change the layout, and this one could become ADR 6.

- We mention how we're redefining the send/delivery distinction three times (in each section). The one I find confusing is in "Specifics", because it's really a consequence. Much of what is said in that section could be folded into other sections, also without too much effort (I hope).

I'm hoping those suggestions resonate with you, as they're not really about the content itself. Let's have a chat if the "movements" I'm suggesting aren't clear.

**Comment by kevindew** [comment]
@benthorner it's a bit tedious having you ask for a restructure after it's already been through one round of amends. Do you want to open a PR to change the ADR structure for 5 and I'll amend this to be ADR 6?

**Comment by benthorner** [comment]
Sure, happy to help: https://github.com/alphagov/email-alert-api/pull/1445.

I agree it's not fun to get a second review with major changes. I only suggested these changes because: I think they're quite easy, yet they'll have a big impact on readability. In my original review, I was struggling to picture the outcome of my suggestions and wrap my head around all the content, so I was reading each paragraph in isolation from the rest. The changes you've made have made it much easier for me to read them all together, hence the new comments.

I agree it's not worth spending much more time on this, beyond restructuring 👍.

File: docs/adr/adr-006-email-delivery-responsibilities.md
Comment (benthorner) [inline docs/adr/adr-006-email-delivery-responsibilities.md:29]:
```suggestion

```

Minor: we've said this first sentence already in the previous paragraph.

File: docs/adr/adr-006-email-delivery-responsibilities.md
Comment (benthorner) [inline docs/adr/adr-006-email-delivery-responsibilities.md:32]:
```suggestion
responsibilities of email delivery.
```

Minor: we repeat this decision in the next paragraph (in the decision section), so we don't need to say it here.

File: docs/adr/adr-006-email-delivery-responsibilities.md
Comment (benthorner) [inline docs/adr/adr-006-email-delivery-responsibilities.md:101]:
```suggestion
period, it can be added later to meet any later needs that are identified. The
```

**Review by benthorner — APPROVED** [review]
Thanks, I find it much easier to read with the new structure 🏅. I think there are a couple of sentences we could loose, as we seem to be saying the same thing twice, but that's a minor issue.

File: docs/adr/adr-006-email-delivery-responsibilities.md
Comment (kevindew) [reply docs/adr/adr-006-email-delivery-responsibilities.md]: Oh darn - that's a replacement paragraph that got left behind. I recall this is the second time I've done this recently 🤦

**Pull request merged**
